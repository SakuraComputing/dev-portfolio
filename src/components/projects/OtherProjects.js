import React from 'react';
import OtherProjectItem from '../../components/projects/OtherProjectItem';

class OtherProjects extends React.Component {

    render() {

        const { otherprojects } = this.props;

        let otherProjectItems;

        if(otherprojects.length > 0) {
            otherProjectItems = otherprojects.map(otherproject => (
                <OtherProjectItem key={otherproject.id} otherproject={otherproject} />
            ));
        }

        return(
            <section className="projects section">
                <div className="section-inner">
                    <h2 className="heading">Other Projects</h2>
                    <div className="content">
                        {otherProjectItems}
                        <p></p>
                        <a className="btn btn-cta-secondary" href="#">More on CoderWall <i className="fa fa-chevron-right"></i></a>
                    </div>
                </div>
            </section>
        )
    }
}
export default OtherProjects;

