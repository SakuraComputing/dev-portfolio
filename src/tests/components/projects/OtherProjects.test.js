import React from 'react';
import { shallow } from 'enzyme';
import OtherProjects from '../../../components/projects/OtherProjects';

describe('Other Projects Section', () => {
    it('should render correctly', () => {
        const otherprojects = jest.fn();
        const wrapper = shallow(<OtherProjects otherprojects={otherprojects}/>);
        expect(wrapper).toMatchSnapshot();
    });
});