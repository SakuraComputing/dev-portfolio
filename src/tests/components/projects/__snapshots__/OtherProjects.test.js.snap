// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other Projects Section should render correctly 1`] = `
<section
  className="projects section"
>
  <div
    className="section-inner"
  >
    <h2
      className="heading"
    >
      Other Projects
    </h2>
    <div
      className="content"
    >
      <p />
      <a
        className="btn btn-cta-secondary"
        href="#"
      >
        More on CoderWall 
        <i
          className="fa fa-chevron-right"
        />
      </a>
    </div>
  </div>
</section>
`;
